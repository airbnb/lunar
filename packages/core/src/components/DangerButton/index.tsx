import React from 'react';
import Button, { Props } from '../Button';

const StyledDangerButton = Button.extendStyles(({ color }) => ({
  button: {
    backgroundColor: color.core.danger[5],
    border: `2px solid ${color.core.danger[5]}`,

    '@selectors': {
      ':not([disabled]):hover': {
        backgroundColor: color.core.danger[6],
        borderColor: color.core.danger[6],
      },
    },
  },

  button_inverted: {
    color: color.core.danger[5],
    backgroundColor: color.accent.bg,

    '@selectors': {
      ':not([disabled]):hover': {
        color: color.core.danger[6],
        backgroundColor: color.accent.bgHover,
      },
    },
  },
}));

/** A button to use for primary actions. Supports all the same props as `Button`. */
export default function DangerButton({ children, ...props }: Props) {
  return <StyledDangerButton {...props}>{children}</StyledDangerButton>;
}
