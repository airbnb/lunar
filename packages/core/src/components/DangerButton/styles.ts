import Core from '../..';
import { StyleSheet } from '../../hooks/useStyles';
import baseStyleSheet from '../Button/styles';

const styleSheet: StyleSheet = Core.aesthetic.extendStyles(baseStyleSheet, ({ color }) => ({
  button: {
    backgroundColor: color.core.danger[5],
    border: `2px solid ${color.core.danger[5]}`,

    '@selectors': {
      ':not([disabled]):hover': {
        backgroundColor: color.core.danger[6],
        borderColor: color.core.danger[6],
      },
    },
  },

  button_inverted: {
    color: color.core.danger[5],
    backgroundColor: color.accent.bg,

    '@selectors': {
      ':not([disabled]):hover': {
        color: color.core.danger[6],
        backgroundColor: color.accent.bgHover,
      },
    },
  },
}));

export default styleSheet;
