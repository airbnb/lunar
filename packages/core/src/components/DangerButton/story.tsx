import React from 'react';
import { action } from '@storybook/addon-actions';
import DangerButton from '.';

export default {
  title: 'Core/DangerButton',
  parameters: {
    inspectComponents: [DangerButton],
  },
};

export function buttonWithAnInvertedVariant() {
  return (
    <>
      <DangerButton>Button</DangerButton> <DangerButton inverted>Button</DangerButton>
    </>
  );
}

buttonWithAnInvertedVariant.story = {
  name: 'Button with an inverted variant.',
};

export function anAnchorLinkWhenPassingHref() {
  return (
    <>
      <DangerButton openInNewWindow href="https://github.com/airbnb/lunar">
        Link
      </DangerButton>{' '}
      <DangerButton openInNewWindow inverted href="https://github.com/airbnb/lunar">
        Link
      </DangerButton>
    </>
  );
}

anAnchorLinkWhenPassingHref.story = {
  name: 'An anchor link when passing `href`.',
};

export function withEventHandlers() {
  return (
    <DangerButton
      onClick={action('onClick')}
      onMouseOver={action('onMouseOver')}
      onFocus={action('onFocus')}
    >
      Button
    </DangerButton>
  );
}

withEventHandlers.story = {
  name: 'With event handlers.',
};

export function withDifferentSizingSmallRegularDefaultAndLarge() {
  return (
    <>
      <DangerButton small>Button</DangerButton> <DangerButton>Button</DangerButton>{' '}
      <DangerButton large>Button</DangerButton>
      <br />
      <br />
      <DangerButton inverted small>
        Button
      </DangerButton>{' '}
      <DangerButton inverted>Button</DangerButton>{' '}
      <DangerButton inverted large>
        Button
      </DangerButton>
    </>
  );
}

withDifferentSizingSmallRegularDefaultAndLarge.story = {
  name: 'With different sizing: small, regular (default), and large.',
};

export function withDifferentStatesDisabledAndLoading() {
  return (
    <>
      <DangerButton disabled>Button</DangerButton> <DangerButton loading>Button</DangerButton>
      <br />
      <br />
      <DangerButton inverted disabled>
        Button
      </DangerButton>{' '}
      <DangerButton inverted loading>
        Button
      </DangerButton>
    </>
  );
}

withDifferentStatesDisabledAndLoading.story = {
  name: 'With different states: disabled and loading.',
};
