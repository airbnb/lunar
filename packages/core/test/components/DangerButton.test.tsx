import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import DangerButton from '../../src/components/DangerButton';

describe('<DangerButton />', () => {
  it('renders a primary button', () => {
    const wrapper = shallowWithStyles(<DangerButton>Button</DangerButton>).dive();

    expect(wrapper).toMatchSnapshot();
  });
});
