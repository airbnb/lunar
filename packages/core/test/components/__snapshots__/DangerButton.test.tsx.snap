// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DangerButton /> renders a primary button 1`] = `
<ButtonOrLink
  aria-busy={false}
  className="button_1aptcl0-o_O-button_regular_un5ukp"
>
  Button
</ButtonOrLink>
`;
